src/Fixer/Preload/PreloadExplicitClassSymbolsFixer.php

```diff
@@ -0,0 +1,365 @@
+            if (T_PUBLIC !== $methodAttributes['visibility']) {
+                continue;
+            }
+
+            // Get argument types
+            $arguments = $this->functionsAnalyzer->getFunctionArguments($tokens, $functionIndex);
+            foreach ($arguments as $argument) {
+                if ($argument->hasTypeAnalysis() && !$argument->getTypeAnalysis()->isReservedType()) {
+                    $classes[] = $argument->getTypeAnalysis()->getName();
+                }
+            }
+
+            // Get return type
+            $returnType = $this->functionsAnalyzer->getFunctionReturnType($tokens, $functionIndex);
+            if (null !== $returnType && !$returnType->isReservedType()) {
+                $classes[] = $returnType->getName();
+            }
+        }
+
+        // TODO parse public properties
+        // TODO Get the class' interfaces and parent
+
+        return array_unique($classes);
+    }
+
+    /**
+     * Find a function in the tokens.
+     *
+     * @param string $name
+     *
+     * @return null|int the index or null. The index is to the "function" token.
+     */
+    private function findFunction(Tokens $tokens, $name)
+    {
+        foreach ($tokens as $index => $token) {
+            if (!$token->isGivenKind(T_FUNCTION)) {
+                continue;
+            }
+
+            $nextTokenIndex = $tokens->getNextMeaningfulToken($index);
+            $nextToken = $tokens[$nextTokenIndex];
+
+            if ($nextToken->getContent() !== $name) {
+                continue;
+            }
+
+            return $index;
+        }
+
+        return null;
+    }
+
+    /**
+     * Inject "class_exists" at the top of the file.
+     */
+    private function injectClasses(Tokens $tokens, array $classes)
+    {
+        $insertAfter = null;
+        foreach ($tokens as $index => $token) {
+            if (!$token->isGivenKind(T_CLASS)) {
+                continue;
+            }
+
+            $insertAfter = $tokens->getPrevMeaningfulToken($index);
+
+            break;
+        }
+
+        if (null === $insertAfter) {
+            return;
+        }
+
+        $newTokens = [];
+        foreach ($classes as $class) {
+            $newTokens[] = new Token([T_STRING, 'class_exists']);
+            $newTokens[] = new Token('(');
+
+            $parts = explode('\\', $class);
+            $lastPart = array_pop($parts);
+            foreach ($parts as $part) {
+                if (!empty($part)) {
+                    $newTokens[] = new Token([T_STRING, $part]);
+                }
+                $newTokens[] = new Token([T_NS_SEPARATOR, '\\']);
+            }
+            $newTokens[] = new Token([T_STRING, $lastPart]);
+
+            $newTokens[] = new Token([T_DOUBLE_COLON, '::']);
+            $newTokens[] = new Token([CT::T_CLASS_CONSTANT, 'class']);
+            $newTokens[] = new Token(')');
+            $newTokens[] = new Token(';');
+            $newTokens[] = new Token([T_WHITESPACE, "\n"]);
```

Comment:
```suggestion
            $newTokens[] = new Token([T_WHITESPACE, $this->whitespacesConfig->getLineEnding()]);
```

(you need to implement WhitespacesAwareFixerInterface also)